pages/finance.tsx

```diff
@@ -20,7 +21,15 @@ function Finance() {
   const startDate = dateBeforeDays(3)
   const endDate = dateAfterDays(3)
 
-  return <div>{user ? <FinanceDateHeader startDate={startDate} endDate={endDate} /> : null}</div>
+  if (!user) {
```

Comment:
instead have an auth guard in the router so that the user can never reach this page if not authenticated

Reply:
In the auth guard check that if user not logged in then take user to login page